client/src/components/views/Shuttles/index.js

```diff
@@ -130,45 +151,153 @@ class ShuttleBay extends Component {
       clamps,
       compress,
       doors,
-      damage
+      damage,
+      direction
     } = this.props;
     const { animating } = this.state;
+
+    let hint = { clamps: '', compress: '', doors: '' };
+
+    let shuttleStatusMsg = '';
+    if(direction === 'departing' && docked) {
+      shuttleStatusMsg = 'Preparing shuttle for departure.';
+      
+      if(!clamps) hint.clamps = 'ok';
+      if(!compress) hint.compress = 'ok';
+      if(!doors) hint.doors = 'ok';
+
+      if(clamps) hint.clamps = 'attention';
+      else if(compress) hint.compress = 'attention';
+      else if(doors) hint.doors = 'attention';
+      else shuttleStatusMsg = 'Shuttle is departing.';
+    }
+    else if (direction === 'departing' && !docked) {
+      shuttleStatusMsg = 'Shuttle has departed. Secure shuttle bay.';
+
+      if(!clamps) hint.clamps = 'ok';
+      if(compress) hint.compress = 'ok';
+      if(doors) hint.doors = 'ok';
+      
+      if(!doors) hint.doors = 'attention';
+      else if(!compress) hint.compress = 'attention';
+      else shuttleStatusMsg = 'Shuttle bay secured.';
+    }
+    else if (direction === 'arriving' && !docked) {
+      shuttleStatusMsg = 'Shuttle approaching. Prepare for arrival.';
+
+      if(!clamps) hint.clamps = 'ok';
+      if(!compress) hint.compress = 'ok';
+      if(!doors) hint.doors = 'ok';
+
+      if(clamps) hint.clamps = 'attention';
+      else if(compress) hint.compress = 'attention';
+      else if(doors) hint.doors = 'attention';
+      else shuttleStatusMsg = 'The shuttle is entering the bay. Please wait.';
+    }
+    else if (direction === 'arriving' && docked) {
+      shuttleStatusMsg = 'Shuttle has arrived. Please secure shuttle.';
+
+      if(clamps) hint.clamps = 'ok';
+      if(compress) hint.compress = 'ok';
+      if(doors) hint.doors = 'ok';
+
+      if(!doors) hint.doors = 'attention';
+      else if(!compress) hint.compress = 'attention';
+      else if(!clamps) hint.clamps = 'attention';
+      else shuttleStatusMsg = 'Shuttle secured.';
+    }
+    else if (!docked) {
+      if(!clamps) hint.clamps = 'ok';
+      if(compress) hint.compress = 'ok';
+      if(doors) hint.doors = 'ok';
+    }
+    else if (docked) {
+      if(clamps) hint.clamps = 'ok';
+      if(compress) hint.compress = 'ok';
+      if(doors) hint.doors = 'ok';
+    }
+
+    var hintStrings = {
+      'ok': '\u2713',
```

Comment:
Can we switch this to using the `<FontAwesome />` component from `react-fontawesome`? I can't guarantee that these unicode icons will be consistent across platforms, but Font Awesome will pretty much always look the same.

Reply:
I changed them, and FontAwesome looks a lot better than unicode anyway.